### PR TITLE
stunglasses

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

sunglasses with flashes in them.

## Why It's Good For The Game

I dunno. I thought of it and thought it was fun and thought "why not"

## Changelog
:cl:
add: stunglasses
/:cl:

### CODE REVIEW

**Comment by Seris02** [comment]
I wanted to see if :honk: would show up in OOC chat for the PR thing when it's closed and open. It doesn't.


Sad.

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -247,6 +247,36 @@
...
+		installed = new(src)
+
+/obj/item/clothing/glasses/sunglasses/stunglasses/ui_action_click(mob/user)
+	if (installed && !installed.crit_fail)
+		installed.attack_self(user)
+	else
+		to_chat(user, "<span class = 'danger'>Install a new flash in [src]!</span>")
+
+
+/obj/item/clothing/glasses/sunglasses/stunglasses/attackby(obj/item/W,mob/user)
```
Comment (silicons) [inline code/modules/clothing/glasses/_glasses.dm]: needs to call . = ..() if possible

File: code/modules/clothing/glasses/_glasses.dm [hunk 2]
```diff
@@ -247,6 +247,36 @@
...
+	if (!installed)
+		installed = new(src)
+
+/obj/item/clothing/glasses/sunglasses/stunglasses/ui_action_click(mob/user)
+	if (installed && !installed.crit_fail)
+		installed.attack_self(user)
+	else
+		to_chat(user, "<span class = 'danger'>Install a new flash in [src]!</span>")
+
+
```
Comment (silicons) [inline code/modules/clothing/glasses/_glasses.dm]: extra line

File: code/modules/clothing/glasses/_glasses.dm [hunk 3]
```diff
@@ -247,6 +247,36 @@
...
 	glass_colour_type = /datum/client_colour/glass_colour/red
 
+/obj/item/clothing/glasses/sunglasses/stunglasses
+	name = "stunglasses"
+	desc = "Sunglasses with inbuilt flashes. Made for those who prefer to walk around in style, who needs clumsy flashes anyway?"
+	actions_types = list(/datum/action/item_action/flash)
+	var/obj/item/assembly/flash/installed
+
+/obj/item/clothing/glasses/sunglasses/stunglasses/Initialize()
+	..()
```
Comment (silicons) [inline code/modules/clothing/glasses/_glasses.dm]: . = ..()

File: code/modules/crafting/recipes/recipes_clothing.dm [hunk 4]
```diff
@@ -87,6 +87,16 @@
 	reqs = list(/obj/item/clothing/glasses/sunglasses/reagent = 1)
 	category = CAT_CLOTHING
 
+/datum/crafting_recipe/stunglasses
```
Comment (silicons) [inline code/modules/crafting/recipes/recipes_clothing.dm:90]: maybe no crafting recipe for now.

Comment (Seris02) [reply code/modules/crafting/recipes/recipes_clothing.dm:90]: sad.

**Review by Linzolle — APPROVED** [review]
seems pretty harmless especially if there's no crafting recipe.

**Comment by Ghommie** [comment]
Resolve conflicts and then I'll merge this silly stuff.

**Pull request merged**